src/engine/world/room.cpp

```diff
@@ -584,12 +586,39 @@ void Room::collectShaderLights(size_t depth)
   lightsBuffer->setData(bufferLights, gl::api::BufferUsage::StaticDraw);
 }
 
-void Room::createParticleMesh(const std::string& label,
-                              const glm::vec3& min,
-                              const glm::vec3& max,
-                              render::scene::MaterialManager& materialManager)
+void Room::regenerateDust(const std::shared_ptr<engine::Presenter>& presenter,
+                          const gslu::nn_shared<render::scene::Material>& dustMaterial,
+                          bool isDustEnabled,
+                          uint8_t dustDensityDivisor)
 {
-  static const constexpr auto Resolution = (1_sectors / 12).cast<float>().get();
+  if(!isDustEnabled)
+  {
+    dust = nullptr;
+    return;
+  }
+
+  if(const auto it = dustCache.find(dustDensityDivisor); it != dustCache.end())
+  {
+    dust = it->second;
+    return;
+  }
+  if(presenter != nullptr)
+    presenter->drawLoadingScreen(_("Generating Dust Particles..."));
+  auto dustNode
+    = createParticleMesh(node->getName() + "/dust", verticesBBoxMin, verticesBBoxMax, dustMaterial, dustDensityDivisor);
+  dustCache.emplace(dustDensityDivisor, dustNode);
+  dust = dustNode;
+}
+
+std::shared_ptr<render::scene::Node>
+  Room::createParticleMesh(const std::string& label,
+                           const glm::vec3& min,
+                           const glm::vec3& max,
+                           const gslu::nn_shared<render::scene::Material>& dustMaterial,
+                           uint8_t dustDensityDivisor)
+{
+  static const constexpr auto BaseGridAxisSubdivision = 12;
+  static const auto resolution = (cbrt(dustDensityDivisor) / BaseGridAxisSubdivision * 1_sectors).cast<float>().get();
```

Comment:
Sorry, I completely missed this throughout reviewing and testing... the `static` keyword resulted in `resolution` only calculated once and then never again, fixed this on `master`.

Reply:
Good Catch. The real time update would have been for nothing if you hadn't found this.